src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs

```diff
@@ -68,5 +96,293 @@ protected static SymbolDisplayPart NewLine()
+                    }
+                    else
+                    {
+                        foundParameterIndex = FirstUnspecifiedParameter(argToParamMap, argumentCount);
+                    }
+                }
+                else
+                {
+                    foundParameterIndex = -1;
+                }
+
+                Debug.Assert(foundParameterIndex < parameters.Length);
+
+                return true;
+
+                // If the cursor is pointing at an argument for which we did not find the corresponding
+                // parameter, we will highlight the first unspecified parameter.
+                static int FirstUnspecifiedParameter(ArrayBuilder<int> argToParamMap, int argumentCount)
+                {
+                    using var _ = ArrayBuilder<bool>.GetInstance(argumentCount, false, out var specified);
+                    for (var i = 0; i < argumentCount; i++)
+                    {
+                        var parameterIndex = argToParamMap[i];
+                        if (parameterIndex >= 0)
+                        {
+                            specified[parameterIndex] = true;
+                        }
+                    }
+
+                    var first = specified.FindIndex(s => !s);
+                    return first <= 0 ? 0 : first;
+                }
+
+                // Determines if the given argument is compatible with the given parameter
+                bool IsCompatibleArgument(ArgumentSyntax argument, IParameterSymbol parameter)
+                {
+                    var parameterRefKind = parameter.RefKind;
+                    if (parameterRefKind == RefKind.None)
+                    {
+                        if (IsEmptyArgument(argument.Expression))
+                        {
+                            // An argument left empty is considered to match any parameter
+                            // M(1, $$)
+                            // M(1, , 2$$)
+                            return true;
+                        }
+
+                        var type = parameter.Type;
+                        if (parameter.IsParams
+                            && type is IArrayTypeSymbol arrayType
+                            && HasImplicitConversion(argument.Expression, arrayType.ElementType))
+                        {
+                            return true;
+                        }
+
+                        return HasImplicitConversion(argument.Expression, type);
+                    }
+
+                    var argumentRefKind = argument.GetRefKind();
+                    if (parameterRefKind == RefKind.In && argumentRefKind == RefKind.None)
+                    {
+                        // A by-value argument matches an `in` parameter
+                        return true;
+                    }
+
+                    if (parameterRefKind == argumentRefKind)
+                    {
+                        return true;
+                    }
+
+                    return true;
```

Comment:
```suggestion
                    if (parameterRefKind == argumentRefKind)
                    {
                        return true;
                    }

                    return true;
```

this is the same as just `return true;`?

Reply:
There was a bug there. Fixed and added a test. Thanks!